mediaire_toolbox/data_cleaner.py

```diff
@@ -12,41 +14,129 @@
-time it will be cleaned up. 
+time it will be cleaned up.
 """
 
 
 class DataCleaner:
 
-    def __init__(self, folder, max_folder_size, max_data_seconds):
+    def __init__(self, folder, max_folder_size, max_data_seconds,
+                 whitelist=None, blacklist=None):
+        """
+        Parameters
+        ----------
+        folder:
+            folder: str
+            Path of folder to be cleaned.
+        max_folder_size: int
+            Max folder size, delete until current size is smaller than this.
+            -1 if not deleting files based on size.
+        max_data_seconds: int
+            Max data age, delete folders that are older than this many seconds.
+            -1 if not deleting files based on age.
+        whitelist: list
+            Whitelist for files. List of Unix like filename pattern strings.
+        blacklist: list
+            Blacklist for files. List of Unix like filename pattern strings.
+        """
         self.base_folder = folder
         self.max_folder_size = max_folder_size
         self.max_data_seconds = max_data_seconds
+        self.whitelist = whitelist
+        self.blacklist = blacklist
+        if whitelist and blacklist:
+            raise ValueError("Both black list and white list in instance")
         default_logger.info("""Instantiated a DataCleaner on folder {%s} with max 
                     foldersize={%s} and max data seconds={%s}""" %
                     (folder, max_folder_size, max_data_seconds))
 
-    def current_size(self):
+    @staticmethod
+    def current_size(folder):
         """disk usage in bytes"""
-        return int(subprocess.check_output(['du', '-s', self.base_folder]) \
-            .split()[0] \
-            .decode('utf-8'))
+        if os.path.isdir(folder):
+            return int(subprocess.check_output(['du', '-s', folder]) \
+                .split()[0] \
+                .decode('utf-8'))
+        else: 
+            return 0
 
-    def creation_time(self, folder):
+    @staticmethod
+    def creation_time(folder):
         return int(os.path.getmtime(folder))
 
     @staticmethod
     def list_sub_folders(folder):
         return [os.path.join(folder, sub_folder)
                 for sub_folder in os.listdir(folder)]
 
+    def clean_folder(self, folder, dry_run=False):
+        """Cleans the contents of the parameter folder. 
+        If the instance was configured with a whitelist or a blacklist, 
+        a selective delete will be performed accordingly."
+
+        Returns
+        -------
+        list
+            Returns the list of files deleted.
+        """
+        
+        # get all folders and files in the folder
+        file_list, folder_list = [], [folder]
+        for root, directories, filenames in os.walk(folder, topdown=False):
+            file_list = file_list + [os.path.join(root, filename)for filename in filenames]
+            folder_list = folder_list + [os.path.join(root, directory)for directory in directories]
```

Comment:
could these two expressions be written as += ? also beware of formatting here, it seems way above 79 columns